chapter2task6

DeaDMop03 opened the pull request.

DeaDMop03 committed:
chapter2task6 release

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,43 @@ public static void main(String[] args) {
+            return 0;
+        }
+
+            while (m != n) {
```
через рекурсию пожалуйста

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,43 @@ public static void main(String[] args) {
+        int greaterNumber = Math.max(m, n);
+        int smallerNumber = Math.min(m, n);
+        int noc = greaterNumber;
+        while (noc % smallerNumber != 0) {
```
блоки кода отделяют пустыми строками, если они стоят не в сразу после { и не перед }

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,43 @@ public static void main(String[] args) {
+        int smallerNumber = Math.min(m, n);
+        int noc = greaterNumber;
+        while (noc % smallerNumber != 0) {
+            noc += noc;
```
неверный принцип, каждая итерация цикла будет увеличивать число в 2 раза.

DeaDMop03 committed:
chapter2task6
fixed getNoc (changed increment of noc)
fixed getNodByEuclideanAlgorithm (added recursion)

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,45 @@ public static void main(String[] args) {
+
+            if (m != n) {
+                if (m > n) {
+                    m = m - n;
```
не используйте разность в методе, пользуйте %

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,45 @@ public static void main(String[] args) {
     static int getNodByEuclideanAlgorithm(int m, int n){
         // Ваш код
-        return 0;
+        if (m == 0 || n == 0) {
```
уберите это условие из метода

DeaDMop03 committed:
chapter2task6 fix (changed  substraction to modulo)

EvgeneZhurov commented on src/main/java/com/walking/intensive/chapter2/task6/Task6.java:
```diff
@@ -9,18 +9,45 @@ public static void main(String[] args) {
+            return 0;
+        }
+
+            if (m != n) {
```
если m % n == 0, то верните b

return рекурсия с аргументами b , m % n

пробуйте.

EvgeneZhurov commented:
висит больше недели.
